DTS: OpenISA RV32M1: Convert driver to new DT_INST/DT_NODELABEL macros

galak opened the pull request.

galak committed:
soc: riscv: openisa_rv32m1: Convert from Kconfig to DT_NODELABEL

As prep for drivers being converted to utilize DT_INST and removal of
per instance Kconfig symbols, move soc.c code to utilize DT_NODELABEL
instead.

Also rename various node labels to match the SoC docs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: riscv: rv32m1_vega: Convert from Kconfig to DT_NODELABEL

As prep for drivers being converted to utilize DT_INST and removal of
per instance Kconfig symbols, move board pinmux.c code to utilize
DT_NODELABEL instead.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: i2c: i2c_rv32m1_lpi2c: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros and remove related board per
instance Kconfig symbol usage.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: intc: rv32m1_intmux: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: serial: rv32m1_lpuart: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros and remove related board per
instance Kconfig symbol usage.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
soc: riscv: openisa_rv32m1: Add helper to get clock_ip_name from dts

Add a helper macro that will go from a DT_DRV_INST number and return the
clock_ip_name value for that instance.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: spi: rv32m1_spi: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros and remove related board per
instance Kconfig symbol usage.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: timer: rv32m1_lptmr: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: pwm: rv32m1_tpm: Use DT_INST_FOREACH

Convert driver to use DT_INST_FOREACH(TPM_DEVICE)

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: riscv: rv32m1_vega: remove stale Kconfig PWM setting

Per instance Kconfig symbols aren't used by the rv32m1 lpspi driver
so remove the unnecessary setting of PWM_2.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: openisa: rv32ma: Update openisa,rv32m1-gpio binding for ports

Add a property to the openisa,rv32m1-gpio binding that relates the GPIO
node to the pinmux PORT node.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: gpio: rv32m1: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros and remove related board per
instance Kconfig symbol usage.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

pabigot reviewed (changes requested):
One suggested change.

NAK because this appears to break `rv32m1_vega_ri5cy`.  hello_world and blinky work as expected on master, but do nothing on this branch (as-is, or rebased on master)

galak commented on drivers/serial/uart_rv32m1_lpuart.c:
```diff
@@ -294,166 +296,54 @@ static const struct uart_driver_api rv32m1_lpuart_driver_api = {
+		.clock_ip_src = kCLOCK_IpSrcFircAsync,			\
+		.baud_rate = DT_INST_PROP(n, current_speed),		\
+		.hw_flow_control = DT_INST_PROP(n, hw_flow_control),	\
+		IRQ_FUNC_INIT						\
```
The problem with that is irq_config_func member only exists if CONFIG_UART_INTERRUPT_DRIVEN is defined, so that doesn't work.

pabigot replied:
Ick.

galak committed:
drivers: pinmux: rv32m1: Convert driver to new DT_INST macros

Convert driver to use DT_INST macros and remove related board per
instance Kconfig symbol usage.

Updated the openisa,rv32m1_vega-pinmux binding to require the label
property and updated the rv32m1.dtsi to add label properties for the
pinmux nodes.

Also update gpio_basic_api test to use DT_NODELABEL.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
> One suggested change.
> 
> NAK because this appears to break `rv32m1_vega_ri5cy`. hello_world and blinky work as expected on master, but do nothing on this branch (as-is, or rebased on master)

Fixed issue, there was a typo in the 'INST_DT_CLOCK_IP_NAME' macro:

```
diff --git a/soc/riscv/openisa_rv32m1/soc.h b/soc/riscv/openisa_rv32m1/soc.h
index 5a10ff6c31..60ad1267c9 100644
--- a/soc/riscv/openisa_rv32m1/soc.h
+++ b/soc/riscv/openisa_rv32m1/soc.h
@@ -111,6 +111,6 @@ void soc_interrupt_init(void);
 /* helper macro to convert from a DT_INST to HAL clock_ip_name */
 #define INST_DT_CLOCK_IP_NAME(n) \
        MAKE_PCC_REGADDR(DT_REG_ADDR(DT_INST_PHANDLE(n, clocks)), \
-                       DT_INST_CLOCKS_CELL(n, name))
+                       DT_INST_CLOCKS_CELL(0, name))
 ```

pabigot reviewed (approved):
Assuming the only change outside the rebase is the fix for clock this should be fine.  GPIO driver test passes.
